tests/contract_testcases/evm/error.sol

```diff
@@ -0,0 +1,38 @@
+
+error E1(int bar, bool);
+error E2(int bar, bool foo);
+
+contract e {
+	error E3(int bar, bool foo);
+}
+
+contract c is e {
+	error E4(int bar, bool foo);
+
+	function named1() public {
+		if (1 != 0) {
+			revert E1({bar: 1, foo: true});
+		} else if (1 != 0) {
+			revert E2({bar: 1});
+		} else if (1 != 0) {
+			revert E3({bar: 1, foo: true, baz: "hey"});
+		} else if (1 != 0) {
```

Comment:
Rustc would complain here that all the branches check the same condition. If I am not wrong, clang would do the same!

Reply:
Perhaps this is something we can do too (in the future).